File: src/main/java/org/openbase/planetsudo/jp/JPServerHostname.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openbase.planetsudo.jp;

import org.openbase.jps.preset.AbstractJPString;


/**
 *
 * @author Divine Threepwood
 */
public class JPServerHostname extends AbstractJPString {

	public final static String[] COMMAND_IDENTIFIERS = {"--server"};
	public final static String[] ARGUMENT_IDENTIFIER = {"HOSTNAME"};

	public JPServerHostname() {
		super(COMMAND_IDENTIFIERS);
	}
    
    @Override
    protected String[] generateArgumentIdentifiers() {
        return ARGUMENT_IDENTIFIER;
    }
	
	@Override
	protected String getPropertyDefaultValue() {
		return "localhost";
	}

	@Override
	public String getDescription() {
		return "Setup the default server.";
	}
}

File: src/main/java/org/openbase/planetsudo/jp/JPServerHostname.kt
package org.openbase.planetsudo.jp

import org.openbase.jps.preset.AbstractJPString

/**
 *
 * @author Divine Threepwood
 */
class JPServerHostname : AbstractJPString(COMMAND_IDENTIFIERS) {
    override fun generateArgumentIdentifiers(): Array<String> {
        return ARGUMENT_IDENTIFIER
    }

    override fun getPropertyDefaultValue(): String {
//		return "cpwe";
        return "bone.no-ip.biz"
    }

    override fun getDescription(): String {
        return "Setup the default server."
    }

    companion object {
        val COMMAND_IDENTIFIERS: Array<String> = arrayOf("--server")
        val ARGUMENT_IDENTIFIER: Array<String> = arrayOf("HOSTNAME")
    }
}
